### PR TITLE
refactor(db): optimize db lock

### PR DESCRIPTION
**Pull request by lurais**
**What does this PR do?**
To optimize the way of db lock

**Why are these changes required?**

**This PR has been tested by:**
- Unit Tests
- Manual Testing

**Follow up**

**Extra details**

### CODE REVIEW

File: chainbase/src/main/java/org/tron/common/storage/leveldb/LevelDbDataSourceImpl.java [hunk 1]
```diff
@@ -257,14 +279,17 @@ public Set<byte[]> allKeys() {
   @Override
   public Set<byte[]> allValues() {
     resetDbLock.readLock().lock();
-    try (DBIterator iterator = getDBIterator()) {
-      Set<byte[]> result = Sets.newHashSet();
-      for (iterator.seekToFirst(); iterator.hasNext(); iterator.next()) {
-        result.add(iterator.peekNext().getValue());
+    try {
+      checkDbAlive();
+      try (DBIterator iterator = getDBIterator()) {
```
Comment (halibobo1205) [inline chainbase/src/main/java/org/tron/common/storage/leveldb/LevelDbDataSourceImpl.java:284]: move checkDbAlive into getDBIterator()?

File: chainbase/src/main/java/org/tron/common/storage/rocksdb/RocksDbDataSourceImpl.java [hunk 2]
```diff
@@ -102,37 +106,36 @@ public void closeDB() {
...
     resetDbLock.readLock().lock();
-    Set<byte[]> result = Sets.newHashSet();
-    try (final RocksIterator iter = getRocksIterator()) {
-      for (iter.seekToFirst(); iter.isValid(); iter.next()) {
-        result.add(iter.key());
+    try {
+      checkDbAlive();
+      Set<byte[]> result = Sets.newHashSet();
+      try (final RocksIterator iter = getRocksIterator()) {
+        for (iter.seekToFirst(); iter.isValid(); iter.next()) {
```
Comment (halibobo1205) [inline chainbase/src/main/java/org/tron/common/storage/rocksdb/RocksDbDataSourceImpl.java:134]: same to leveldb

**Review by halibobo1205 — CHANGES_REQUESTED** [review]

File: chainbase/src/main/java/org/tron/common/storage/leveldb/LevelDbDataSourceImpl.java (hunk 1 above)
Comment (lurais) [reply chainbase/src/main/java/org/tron/common/storage/leveldb/LevelDbDataSourceImpl.java:284]: check db shall be before any db operations and just need check once actually for one public method ,if put checkdb operation in a private method , the times of checkDb is not sure

Comment (halibobo1205) [reply chainbase/src/main/java/org/tron/common/storage/leveldb/LevelDbDataSourceImpl.java:284]: getDBIterator() is not a db operation?

Comment (lurais) [reply chainbase/src/main/java/org/tron/common/storage/leveldb/LevelDbDataSourceImpl.java:284]: getDBIterator is a db operation and before this method should checkdb but not in this method as we will not use this private method directly

**Pull request closed without merging**